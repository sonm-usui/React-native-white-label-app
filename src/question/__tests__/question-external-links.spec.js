// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import type { GenericObject } from '../../common/type-common'

import {
  getExternalLinkValidity,
  QuestionExternalLinks,
  QuestionExternalLink,
  MAX_EXTERNAL_LINKS_TO_SHOW,
} from '../components/question-external-links'
import {
  mockExternalLinks,
  mockExternalLink,
} from '../../../__mocks__/data/question-store-mock-data'

describe('fn:getExternalLinkValidity', () => {
  it('should return error if external_links is not an array', () => {
    // We have added bunch of $FlowFixMe inside this test
    // because we want to test validation function by passing
    // different data types and checking for validation result
    // but, since our function is typed specifically for Array<ExternalLink>
    // we can't pass anything else to it, so just function call
    // is spared from flow checks to facilitate testing
    // with multiple different data types
    // $FlowFixMe
    const objectValidation = getExternalLinkValidity({})
    expect(objectValidation).toMatchSnapshot()

    // $FlowFixMe
    const emptyStringValidation = getExternalLinkValidity('')
    expect(emptyStringValidation).toMatchSnapshot()

    // $FlowFixMe
    const stringValidation = getExternalLinkValidity('{}')
    expect(stringValidation).toMatchSnapshot()

    // $FlowFixMe
    const numberValidation = getExternalLinkValidity(0)
    expect(numberValidation).toMatchSnapshot()

    // $FlowFixMe
    const NaNValidation = getExternalLinkValidity(NaN)
    expect(NaNValidation).toMatchSnapshot()

    // $FlowFixMe
    const nullValidation = getExternalLinkValidity(null)
    expect(nullValidation).toMatchSnapshot()

    // $FlowFixMe
    const undefinedValidation = getExternalLinkValidity(undefined)
    expect(undefinedValidation).toMatchSnapshot()
  })

  it('should not return an error if externalLinks array is empty', () => {
    expect(getExternalLinkValidity([])).toMatchSnapshot()
  })

  it('should return an error if link object is not defined correctly', () => {
    // $FlowFixMe Refer FlowFixMe reason from first test description
    const nullLinkValidation = getExternalLinkValidity([null])
    expect(nullLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const stringLinkValidation = getExternalLinkValidity(['string'])
    expect(stringLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const booleanLinkValidation = getExternalLinkValidity([true])
    expect(booleanLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const arrayLinkValidation = getExternalLinkValidity([[true]])
    expect(arrayLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const emptyObjectLinkValidation = getExternalLinkValidity([{}])
    expect(emptyObjectLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const otherPropObjectLinkValidation = getExternalLinkValidity([{ text: 1 }])
    expect(otherPropObjectLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const wrongValueObjectLinkValidation = getExternalLinkValidity([
      { text: '1' },
    ])
    expect(wrongValueObjectLinkValidation).toMatchSnapshot()

    const wrongSrcObjectLinkValidation = getExternalLinkValidity([
      { text: '1', src: '' },
    ])
    expect(wrongSrcObjectLinkValidation).toMatchSnapshot()

    // $FlowFixMe
    const wrongSrcObjectLinkValidation1 = getExternalLinkValidity([
      { text: '1', src: 1 },
    ])
    expect(wrongSrcObjectLinkValidation1).toMatchSnapshot()
  })

  it('should return an error if more than 1000 external link objects are passed', () => {
    const externalLinks = Array.from({ length: 1001 }, (v, index) => ({
      src: `${index}`,
    }))
    expect(getExternalLinkValidity(externalLinks)).toMatchSnapshot()
  })

  it('should not return an error if all link objects are correct', () => {
    expect(
      getExternalLinkValidity([{ text: 'some link', src: 'some link' }])
    ).toMatchSnapshot()
    expect(getExternalLinkValidity([{ src: 'some link' }])).toMatchSnapshot()
  })
})

describe('<QuestionExternalLinks />', () => {
  it('should render validation error if wrong data is passed', () => {
    const { snapshot } = setup({ externalLinks: [{ text: '1' }] })
    expect(snapshot).toMatchSnapshot()
  })

  it('should render links if links are passed', () => {
    const { snapshot } = setup()
    expect(snapshot).toMatchSnapshot()
  })

  it('should render MAX_EXTERNAL_LINKS_TO_SHOW links if more than limit is passed', () => {
    const { snapshot, instance } = setup({
      externalLinks: Array.from({ length: 40 }, () => mockExternalLink),
    })
    expect(snapshot).toMatchSnapshot()

    const externalLinks = instance.findAllByType(QuestionExternalLink)
    expect(externalLinks.length).toBe(MAX_EXTERNAL_LINKS_TO_SHOW)
  })

  it('should render null if no link is passed', () => {
    const { snapshot } = setup({ externalLinks: [] })
    expect(snapshot).toMatchSnapshot()
  })

  function setup(extraProps: ?GenericObject = {}) {
    const props = getProps(extraProps)
    const component = renderer.create(<QuestionExternalLinks {...props} />)
    const instance = component.root
    const snapshot = component.toJSON()

    return { props, component, snapshot, instance }
  }

  function getProps(extraProps: ?GenericObject = {}) {
    return {
      externalLinks: mockExternalLinks,
      ...extraProps,
    }
  }
})
