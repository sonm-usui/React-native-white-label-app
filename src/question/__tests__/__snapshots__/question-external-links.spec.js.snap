// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionExternalLinks /> should render MAX_EXTERNAL_LINKS_TO_SHOW links if more than limit is passed 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
</View>
`;

exports[`<QuestionExternalLinks /> should render links if links are passed 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        An external link
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "System",
            },
            Object {},
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "blue",
            },
          ]
        }
        suppressHighlighting={true}
      >
        https://secondlink.com/external
      </Text>
    </View>
  </View>
</View>
`;

exports[`<QuestionExternalLinks /> should render null if no link is passed 1`] = `null`;

exports[`<QuestionExternalLinks /> should render validation error if wrong data is passed 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontSize": 15,
        },
        Object {
          "fontFamily": "System",
        },
        Object {},
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        Object {
          "color": "#505050",
        },
      ]
    }
    suppressHighlighting={true}
  >
    Invalid external link. Error: CM-QUE-011
  </Text>
</View>
`;

exports[`fn:getExternalLinkValidity should not return an error if all link objects are correct 1`] = `null`;

exports[`fn:getExternalLinkValidity should not return an error if all link objects are correct 2`] = `null`;

exports[`fn:getExternalLinkValidity should not return an error if externalLinks array is empty 1`] = `null`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 1`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 2`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 3`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 4`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 5`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 6`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 7`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 8`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if link object is not defined correctly 9`] = `
Object {
  "code": "CM-QUE-011",
  "message": "One or more link object inside \\"external_links\\" array is invalid. Link object should be of format { text?:string, src:string }, where \\"text\\" property is optional. However, if \\"text\\" property is defined, then it should be a string with less than or equal to 1000 characters. \\"src\\" property should be a string and is not optional.",
}
`;

exports[`fn:getExternalLinkValidity should return an error if more than 1000 external link objects are passed 1`] = `
Object {
  "code": "CM-QUE-012",
  "message": "\\"external_links\\" array should not have more than 1000 link objects.",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 1`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 2`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 3`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 4`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 5`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 6`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;

exports[`fn:getExternalLinkValidity should return error if external_links is not an array 7`] = `
Object {
  "code": "CM-QUE-010",
  "message": "property \\"external_links\\" should be an array of object type { text?:string, src:string }",
}
`;
